### ISSUE TITLE
Functions do not work correctly for languages other than English

### ISSUE THREAD

**Issue by petira** [getgrav/grav#2027]
I found the problem that the `Hyphenize` and `Regex Replace` functions for the Czech language that uses accents are not working correctly. The same will probably apply to other languages that have characters other than English alphabet.

Specifically, the `hyphenize` should convert the `Příliš žluťoučký kůň` to the `prilis-zlutoucky-kun`, but the result is `p-li-lu-ou-k-k-`. This is bad for several reasons, because it is not legible and the dash remains at the end. Also, `Page(s)` is not converted to `page-s`, but to `page-s-`. So if the string is in the system variable `page.header.author`, then name `Tomáš Černý` is converted to `tom-ern-`, although the expected result should be `tomas-cerny`. An example of how to work properly is, for example, converting Page Title to Page Path when creating a page in Admin Plugin.

When I first wanted to remove the accents of `regex_replace`, I encountered the problem that `regex_replace` ignores system variables such as the `page.header.author`, accented characters are completely ignored, and the string remains unchanged. If a string can be set directly, for example `Příliš žluťoučký kůň`, which I want to rewrite (`ř2r`, `í2i`, ...), is the correctly result `Prilis zlutoucky kun`.

I have not yet dealt with other functions.

**Comment by rhukster** [comment]
Yes, this is a known issue.  The original functions were only written for English.  The Grav core language files provide some language regex strings that 'should' allow other languages to be handled, but doing so is beyond my English-only capabilities.

**Comment by petira** [comment]
I understand, but can you change that to work in other languages? I need to generate a URL `[/author]/author-name` from the user name (author of the post) or the author added in the taxonomy `author`.

As I have written, Grav already solves when generating a path from the title page.

http://php.net/manual/en/function.preg-replace.php#96586

or

`i18n-ascii.example.txt`

**Comment by petira** [comment]
@mahagr Following the closing of the issue:

**Positive:**
1. the dash at the end is now removed,
2. `regex_replace` processing system variables such as `page.header.author`.

**Negative:**
1. `hyphenize` still does not work for languages with accents.

**My solution:**
_Page frontmatter:_
```
...
author: 'Příliš žluťoučký kůň'
...
```
_Twig template:_
```
{% set author = page.header.author %}
{{ author }}
{{ author|regex_replace(['/ř/','/í/','/š/','/ž/','/ť/','/č/','/ý/','/ů/','/ň/'], ['r','i','s','z','t','c','y','u','n'])|hyphenize }}
```
_Result:_
Příliš žluťoučký kůň
prilis-zlutoucky-kun

**However, I would prefer if it went directly via `hyphenize` or another character conversion filter, for example, as used when creating a URL from a page title.**

**Comment by mahagr** [comment]
There's no hyphenize library for languages other than English, at least not that I know of. Even Symfony renamed its classes to include word "English"...

**Comment by mahagr** [comment]
@rhukster Looks like there's a new component in Symfony https://github.com/symfony/string

**Comment by rhukster** [comment]
Looks like something that could be added, or maybe just wait for Grav 2.0
